crates/algokit_abi/src/method.rs

```diff
@@ -0,0 +1,550 @@
+        }
+    }
+}
+
+impl Display for ABIReferenceType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let s = match self {
+            ABIReferenceType::Account => "account",
+            ABIReferenceType::Application => "application",
+            ABIReferenceType::Asset => "asset",
+        };
+        write!(f, "{}", s)
+    }
+}
+
+/// Represents the category of an ABI method argument, which can be a value, a transaction, or a reference.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ABIMethodArgType {
+    /// A value that is directly encoded in the application arguments.
+    Value(ABIType),
+    /// A transaction that is placed immediately before the application call in the transaction group.
+    Transaction(ABITransactionType),
+    /// A reference to an account, asset, or application that is encoded as an index into a foreign array.
+    Reference(ABIReferenceType),
+}
+
+impl ABIMethodArgType {
+    /// Check if this is a transaction argument.
+    pub(crate) fn is_transaction(&self) -> bool {
+        matches!(self, ABIMethodArgType::Transaction(_))
+    }
+
+    /// Check if this is a reference argument.
+    pub(crate) fn is_reference(&self) -> bool {
+        matches!(self, ABIMethodArgType::Reference(_))
+    }
+
+    /// Check if this is a value type argument (directly encoded in ApplicationArgs).
+    pub(crate) fn is_value_type(&self) -> bool {
+        matches!(self, ABIMethodArgType::Value(_))
+    }
+}
+
+impl FromStr for ABIMethodArgType {
+    type Err = ABIError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        // Check for direct transaction types first (e.g., "pay", "keyreg", etc.)
+        if let Ok(tx_type) = ABITransactionType::from_str(s) {
+            return Ok(ABIMethodArgType::Transaction(tx_type));
+        }
+
+        // Check for reference types
+        if let Ok(ref_type) = ABIReferenceType::from_str(s) {
+            return Ok(ABIMethodArgType::Reference(ref_type));
+        }
+
+        // Default to ABI value type
+        let abi_type = ABIType::from_str(s)?;
+        Ok(ABIMethodArgType::Value(abi_type))
+    }
+}
+
+/// Represents a parsed ABI method, including its name, arguments, and return type.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ABIMethod {
+    /// The name of the method.
+    pub name: String,
+    /// A list of the method's arguments.
+    pub args: Vec<ABIMethodArg>,
+    /// The return type of the method, or `None` if the method does not return a value.
+    pub returns: Option<ABIType>,
+    /// An optional description of the method.
+    pub desc: Option<String>,
+}
+
+/// Represents an argument in an ABI method.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ABIMethodArg {
+    /// The type of the argument.
+    pub r#type: ABIMethodArgType,
+    /// An optional name for the argument.
+    pub name: Option<String>,
+    /// An optional description of the argument.
+    pub desc: Option<String>,
```

Comment:
What is desc used for? Do we need it?
We've been using the pattern of full names, so potentially this should be `description` if we keep it.